tests/conftest.py

```diff
@@ -117,3 +119,31 @@ def fixture_mock_idref_concept_solver():
     with mock.patch.object(IdRefConceptSolver, "solve") as mock_solve:
         mock_solve.side_effect = fake_idref_concept_solver
         yield mock_solve
+
+
+@pytest.fixture(name="fake_scanr_elastic_client_perform_search")
+def fixture_fake_scanr_elastic_client_perform_search(
+    scanr_api_docs_from_publication,
+    scanr_api_docs_from_person,
+):
+    async def fake_scanr_elastic_client(
+        selected_index: str,
+        base_size: int = 200,
+    ):
+        if selected_index == ScanRElasticClient.Indexes.PUBLICATIONS:
+            for hit in scanr_api_docs_from_publication.get("hits", {}).get("hits", []):
+                yield hit
+        elif selected_index == ScanRElasticClient.Indexes.PERSONS:
+            yield scanr_api_docs_from_person
+        else:
+            raise DereferencingError("Scanr Elastic client not allowed during tests")
```

Comment:
Error message is confusing : replace by "Invalid index requested", as in all valid cases, you serve an answer. So the cause of the exception is not that "Scanr Elastic client is not allowed during tests" (although it is true).